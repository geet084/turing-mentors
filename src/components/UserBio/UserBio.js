import React, { Component } from 'react';
import { CreateInput } from '../CreateInput/CreateInput';

export class UserBio extends Component {
  constructor(props) {
    super(props)
    const { slack, email, phone, location } = this.props
    
    this.state = {
      slack: slack || '',
      email: email || '',
      phone: phone || '',
      location: location || '',
    }
  }

  submitForm = (e) => {
    e.preventDefault();
    this.props.updateUserInfo([this.state, 'userBackground']);
  }

  goBack = (e) => {
    e.preventDefault();
    this.props.updateUserInfo([this.state, 'userInfo']);
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value })
  }

  render() {
    const { slack, email, phone, location } = this.state;
    const mentor = this.props.user === 'Mentor';

    return (
      <div>
        <form onSubmit={this.submitForm} autoComplete='off'>
          <span className="pages">{this.props.user} Contact info:</span>
          <CreateInput field="slack" text="Slack Handle" value={slack} handleChange={this.handleChange} max="28" />
          <CreateInput field="email" text="Email Address" value={email} handleChange={this.handleChange} max="28" />
          <CreateInput field="phone" text="Phone Number" value={phone} handleChange={this.handleChange} max="12" />
          {
            mentor &&
            <CreateInput field="location" text="Location" value={location} handleChange={this.handleChange} max="28" />
          }
        </form>
        <button className="next-btn" onClick={this.goBack}>Back</button>
        <button className="next-btn" onClick={this.submitForm}>Next</button>
        <span className="pages">2 of {mentor ? '6' : '4'}</span>
      </div>
    )
  }
}

export default UserBio;