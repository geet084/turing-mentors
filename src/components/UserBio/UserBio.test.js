import React from 'react';
import { shallow } from 'enzyme';
import { UserBio } from './UserBio';

describe('UserBio', () => {
  let wrapper;
  const mockProps = {
    updateUserInfo: jest.fn()
  };
  const mockState = {
    slack: '',
    email: '',
    phone: '',
    location: '',
  };

  beforeEach(() => {
    wrapper = shallow(<UserBio {...mockProps} />);
  });
  
  describe('initial snapshot', () => {
    it('should match the correct mentee snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
    
    it('should match the correct mentor snapshot', () => {
      wrapper.setProps({ user: 'Mentor' });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('submitForm', () => {
    it('should handle the form submission', () => {
      const mockEvent = { preventDefault: () => { } };
      const expected = [mockState, 'userBackground'];

      wrapper.instance().submitForm(mockEvent);

      expect(mockProps.updateUserInfo).toHaveBeenCalledWith(expected);
    });
  });

  describe('goBack', () => {
    it('should handle going back a page', () => {
      const mockEvent = { preventDefault: () => { } };
      const expected = [mockState, 'userInfo'];

      wrapper.instance().goBack(mockEvent);

      expect(mockProps.updateUserInfo).toHaveBeenCalledWith(expected);
    });
  });

  describe('handleChange', () => {
    it('should update the selected input in state', () => {
      const mockEvent = { target: { name: 'slack', value: 'slack handle' } };
      const expected = 'slack handle';
      expect(wrapper.state('slack')).toEqual('')
      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state('slack')).toEqual(expected);
    });
  });
});