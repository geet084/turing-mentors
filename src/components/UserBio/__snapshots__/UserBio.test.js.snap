// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBio initial snapshot should match the correct mentee snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserBio
    updateUserInfo={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          autoComplete="off"
          onSubmit={[Function]}
        >
          <span
            className="pages"
          >
             Contact info:
          </span>
          <CreateInput
            field="slack"
            handleChange={[Function]}
            max="28"
            text="Slack Handle"
            value=""
          />
          <CreateInput
            field="email"
            handleChange={[Function]}
            max="28"
            text="Email Address"
            value=""
          />
          <CreateInput
            field="phone"
            handleChange={[Function]}
            max="12"
            text="Phone Number"
            value=""
          />
        </form>,
        <button
          className="next-btn"
          onClick={[Function]}
        >
          Back
        </button>,
        <button
          className="next-btn"
          onClick={[Function]}
        >
          Next
        </button>,
        <span
          className="pages"
        >
          2 of 
          4
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "off",
          "children": Array [
            <span
              className="pages"
            >
               Contact info:
            </span>,
            <CreateInput
              field="slack"
              handleChange={[Function]}
              max="28"
              text="Slack Handle"
              value=""
            />,
            <CreateInput
              field="email"
              handleChange={[Function]}
              max="28"
              text="Email Address"
              value=""
            />,
            <CreateInput
              field="phone"
              handleChange={[Function]}
              max="12"
              text="Phone Number"
              value=""
            />,
            false,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " Contact info:",
              ],
              "className": "pages",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " Contact info:",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "field": "slack",
              "handleChange": [Function],
              "max": "28",
              "text": "Slack Handle",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "field": "email",
              "handleChange": [Function],
              "max": "28",
              "text": "Email Address",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "field": "phone",
              "handleChange": [Function],
              "max": "12",
              "text": "Phone Number",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Back",
          "className": "next-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next",
          "className": "next-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2 of ",
            "4",
          ],
          "className": "pages",
        },
        "ref": null,
        "rendered": Array [
          "2 of ",
          "4",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            autoComplete="off"
            onSubmit={[Function]}
          >
            <span
              className="pages"
            >
               Contact info:
            </span>
            <CreateInput
              field="slack"
              handleChange={[Function]}
              max="28"
              text="Slack Handle"
              value=""
            />
            <CreateInput
              field="email"
              handleChange={[Function]}
              max="28"
              text="Email Address"
              value=""
            />
            <CreateInput
              field="phone"
              handleChange={[Function]}
              max="12"
              text="Phone Number"
              value=""
            />
          </form>,
          <button
            className="next-btn"
            onClick={[Function]}
          >
            Back
          </button>,
          <button
            className="next-btn"
            onClick={[Function]}
          >
            Next
          </button>,
          <span
            className="pages"
          >
            2 of 
            4
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "children": Array [
              <span
                className="pages"
              >
                 Contact info:
              </span>,
              <CreateInput
                field="slack"
                handleChange={[Function]}
                max="28"
                text="Slack Handle"
                value=""
              />,
              <CreateInput
                field="email"
                handleChange={[Function]}
                max="28"
                text="Email Address"
                value=""
              />,
              <CreateInput
                field="phone"
                handleChange={[Function]}
                max="12"
                text="Phone Number"
                value=""
              />,
              false,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " Contact info:",
                ],
                "className": "pages",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " Contact info:",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "field": "slack",
                "handleChange": [Function],
                "max": "28",
                "text": "Slack Handle",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "field": "email",
                "handleChange": [Function],
                "max": "28",
                "text": "Email Address",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "field": "phone",
                "handleChange": [Function],
                "max": "12",
                "text": "Phone Number",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "className": "next-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "className": "next-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2 of ",
              "4",
            ],
            "className": "pages",
          },
          "ref": null,
          "rendered": Array [
            "2 of ",
            "4",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`UserBio initial snapshot should match the correct mentor snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserBio
    updateUserInfo={[MockFunction]}
    user="Mentor"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          autoComplete="off"
          onSubmit={[Function]}
        >
          <span
            className="pages"
          >
            Mentor
             Contact info:
          </span>
          <CreateInput
            field="slack"
            handleChange={[Function]}
            max="28"
            text="Slack Handle"
            value=""
          />
          <CreateInput
            field="email"
            handleChange={[Function]}
            max="28"
            text="Email Address"
            value=""
          />
          <CreateInput
            field="phone"
            handleChange={[Function]}
            max="12"
            text="Phone Number"
            value=""
          />
          <CreateInput
            field="location"
            handleChange={[Function]}
            max="28"
            text="Location"
            value=""
          />
        </form>,
        <button
          className="next-btn"
          onClick={[Function]}
        >
          Back
        </button>,
        <button
          className="next-btn"
          onClick={[Function]}
        >
          Next
        </button>,
        <span
          className="pages"
        >
          2 of 
          6
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "off",
          "children": Array [
            <span
              className="pages"
            >
              Mentor
               Contact info:
            </span>,
            <CreateInput
              field="slack"
              handleChange={[Function]}
              max="28"
              text="Slack Handle"
              value=""
            />,
            <CreateInput
              field="email"
              handleChange={[Function]}
              max="28"
              text="Email Address"
              value=""
            />,
            <CreateInput
              field="phone"
              handleChange={[Function]}
              max="12"
              text="Phone Number"
              value=""
            />,
            <CreateInput
              field="location"
              handleChange={[Function]}
              max="28"
              text="Location"
              value=""
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Mentor",
                " Contact info:",
              ],
              "className": "pages",
            },
            "ref": null,
            "rendered": Array [
              "Mentor",
              " Contact info:",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "field": "slack",
              "handleChange": [Function],
              "max": "28",
              "text": "Slack Handle",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "field": "email",
              "handleChange": [Function],
              "max": "28",
              "text": "Email Address",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "field": "phone",
              "handleChange": [Function],
              "max": "12",
              "text": "Phone Number",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "field": "location",
              "handleChange": [Function],
              "max": "28",
              "text": "Location",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Back",
          "className": "next-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next",
          "className": "next-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2 of ",
            "6",
          ],
          "className": "pages",
        },
        "ref": null,
        "rendered": Array [
          "2 of ",
          "6",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            autoComplete="off"
            onSubmit={[Function]}
          >
            <span
              className="pages"
            >
              Mentor
               Contact info:
            </span>
            <CreateInput
              field="slack"
              handleChange={[Function]}
              max="28"
              text="Slack Handle"
              value=""
            />
            <CreateInput
              field="email"
              handleChange={[Function]}
              max="28"
              text="Email Address"
              value=""
            />
            <CreateInput
              field="phone"
              handleChange={[Function]}
              max="12"
              text="Phone Number"
              value=""
            />
            <CreateInput
              field="location"
              handleChange={[Function]}
              max="28"
              text="Location"
              value=""
            />
          </form>,
          <button
            className="next-btn"
            onClick={[Function]}
          >
            Back
          </button>,
          <button
            className="next-btn"
            onClick={[Function]}
          >
            Next
          </button>,
          <span
            className="pages"
          >
            2 of 
            6
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "children": Array [
              <span
                className="pages"
              >
                Mentor
                 Contact info:
              </span>,
              <CreateInput
                field="slack"
                handleChange={[Function]}
                max="28"
                text="Slack Handle"
                value=""
              />,
              <CreateInput
                field="email"
                handleChange={[Function]}
                max="28"
                text="Email Address"
                value=""
              />,
              <CreateInput
                field="phone"
                handleChange={[Function]}
                max="12"
                text="Phone Number"
                value=""
              />,
              <CreateInput
                field="location"
                handleChange={[Function]}
                max="28"
                text="Location"
                value=""
              />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Mentor",
                  " Contact info:",
                ],
                "className": "pages",
              },
              "ref": null,
              "rendered": Array [
                "Mentor",
                " Contact info:",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "field": "slack",
                "handleChange": [Function],
                "max": "28",
                "text": "Slack Handle",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "field": "email",
                "handleChange": [Function],
                "max": "28",
                "text": "Email Address",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "field": "phone",
                "handleChange": [Function],
                "max": "12",
                "text": "Phone Number",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "field": "location",
                "handleChange": [Function],
                "max": "28",
                "text": "Location",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "className": "next-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "className": "next-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2 of ",
              "6",
            ],
            "className": "pages",
          },
          "ref": null,
          "rendered": Array [
            "2 of ",
            "6",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
